scripts/track-dependency-status.sh

```diff
@@ -0,0 +1,47 @@
+# Get total count of python dependencies and outdated python dependencies
+
+total_python_deps="$(pip list| wc -l)"
+outdated_python_deps="$(pip list --outdated| wc -l)"
+
+# Get outdated JS dependency count
+
+function outdated_count {
+    python - <(cat -) <<-END
+import json
+import sys
+with open(sys.argv[1], "r") as json_file:
+    dep_arr = json.load(json_file)
+print(len(dep_arr["data"]["body"]))
+END
+}
+
+outdated_js_deps="$(yarn outdated --json | sed -n 2p | outdated_count)"
+
+
+# Get total JS dependencies from package.json
+
+function dependency_count {
+    local json_path="$1"
+    python - "$json_path" <<-END
+import json
+import sys
+
+with open(sys.argv[1], "r") as json_file:
+    packages = json.load(json_file)
+
+print(sum(len(packages[k]) for k in ["dependencies", "devDependencies"]))
+END
+}
+
+total_js_deps=$(dependency_count package.json)
+
+# Publish metrics to Datadog
+
+source scripts/datadog-utils.sh
+
+send_metric_to_datadog "commcare.static_analysis.dependency.python.total" $total_python_deps "count"
+send_metric_to_datadog "commcare.static_analysis.dependency.python.outdated" $outdated_python_deps "count"
+
+
+send_metric_to_datadog "commcare.static_analysis.dependency.js.total" $total_js_deps "count"
+send_metric_to_datadog "commcare.static_analysis.dependency.js.outdated" $outdated_js_deps "count"
```

Comment:
Just noticed this one. You have "count" here, but I think "gauge" is more appropriate. A gauge metric is when you're measuring a single number over time ("how fast am I going", "how much gas is in the tank", etc); a count metric is when you report "+1" any time a particular thing happens ("how many cars are we passing", "how many times are my windshield wipers swipe?". These metrics here are a single value that we're tracking over time, so I think gauge is more appropriate.

Reply:
Thanks for explaining @dannyroberts, I thought that `count` was used to count an entity over time. I will update the code with "gauge"